### PR TITLE
Add `fjcvtzs` instruction for `ARMv8.3` target

### PR DESCRIPTION
**Pull request by CrazyboyQCD**
Closes https://github.com/boa-dev/boa/issues/4062.

### CODE REVIEW

**Comment by jasonwilliams** [comment]
This is very cool @CrazyboyQCD!

Do you have a way of testing this as I assume Github Actions don't have any arm runners

**Comment by CrazyboyQCD** [comment]
> Do you have a way of testing this as I assume Github Actions don't have any arm runners

`aarch64-apple-darwin` is the only one I found that has the `jsconv` feature, I am not sure if we could use cross to test this.

File: core/engine/src/builtins/number/conversions.rs [hunk 1]
```diff
@@ -72,9 +73,38 @@ pub(crate) fn f64_to_int32(number: f64) -> i32 {
...
+    f64_to_int32(number) as u32
+}
+
 /// Converts a 64-bit floating point number to an `u32` according to the [`ToUint32`][ToUint32] algorithm.
 ///
 /// [ToUint32]: https://tc39.es/ecma262/#sec-touint32
+#[cfg(not(all(target_arch = "aarch64", target_feature = "jsconv")))]
 pub(crate) fn f64_to_uint32(number: f64) -> u32 {
     f64_to_int32(number) as u32
 }
```
Comment (hansl) [inline core/engine/src/builtins/number/conversions.rs:110]:
```suggestion
}


#[test]
fn f64_to_int32_conversion() {
    use crate::builtins::Number;

    assert_eq!(f64_to_int32(0.0), 0);
    assert_eq!(f64_to_int32(-0.0), 0);
    assert_eq!(f64_to_int32(f64::NAN), 0);
    assert_eq!(f64_to_int32(f64::INFINITY), 0);
    assert_eq!(f64_to_int32(f64::NEG_INFINITY), 0);
    assert_eq!(f64_to_int32((i64::from(i32::MAX) + 1) as f64), i32::MIN);
    assert_eq!(f64_to_int32((i64::from(i32::MIN) - 1) as f64), i32::MAX);

    assert_eq!(f64_to_int32(Number::MAX_SAFE_INTEGER + 1.0), 0);
    assert_eq!(f64_to_int32(Number::MIN_SAFE_INTEGER - 1.0), 0);
}
```

**Review by hansl — APPROVED** [review]
It does what it says, and I manually tested it on my Mac, checking both the unit tests and the test262 in debug and release, but also verifying that the outputted release ASM contained the instruction at the right place.

LGTM.

**Review by nekevss — APPROVED** [review]
LGTM! Thanks for looking into this @CrazyboyQCD

**Pull request merged**